File: src/main/java/unaverage/strategic_ench/config/Thorn.java
package unaverage.strategic_ench.config;

import com.typesafe.config.Config;

import java.util.Map;

/**
 * Contains configuration values relating to the thorns enchantment
 */
public class Thorn {
    static final Map<String, Object> DEFAULT = Map.of(
        StopArmorDamage.class.getSimpleName(), StopArmorDamage.DEFAULT
    );

    final StopArmorDamage stopArmorDamage;
    Thorn(Config c) {
        c = c.getConfig(this.getClass().getSimpleName());

        stopArmorDamage = new StopArmorDamage(c);
    }

    /**
     * {@return true if the thorns protection should not damage armor after inflicting damage, false otherwise}
     */
    public boolean stopArmorDamage(){
        return stopArmorDamage.value;
    }

    static class StopArmorDamage {
        static final boolean DEFAULT = true;
        public Boolean value;

        StopArmorDamage(Config c) {
            this.value = c.getBoolean(this.getClass().getSimpleName());
        }
    }
}

File: src/main/kotlin/unaverage/strategic_ench/config/Thorn.kt
package unaverage.strategic_ench.config

import unaverage.strategic_ench.config.Thorn.StopArmorDamage
import java.util.Map

/**
 * Contains configuration values relating to the thorns enchantment
 */
class Thorn internal constructor() {
    val stopArmorDamage: StopArmorDamage

    init {
        stopArmorDamage = StopArmorDamage()
    }

    /**
     * {@return true if the thorns protection should not damage armor after inflicting damage, false otherwise}
     */
    fun stopArmorDamage(): Boolean {
        return stopArmorDamage.value
    }

    class StopArmorDamage {
        var value: Boolean

        init {
            value = DEFAULT
        }

        companion object {
            const val DEFAULT = true
        }
    }

    companion object {
        val DEFAULT = Map.of<String, Any>(
            StopArmorDamage::class.java.simpleName, StopArmorDamage.DEFAULT
        )
    }
}